src/main/default/classes/EspressoMachineTest.cls

```diff
@@ -0,0 +1,54 @@
+@isTest
+class EspressoMachineTest {
+    @isTest
+    static void brewEspresso_RinsedAndHaveWater_ShouldReturnEspresso() {
+        //prepare
+        IEspressoMachine cm = new EspressoMachine();
+        //test
+        Drink Espresso = cm.brewEspresso();
+        //assert
+        System.assertNotEquals(null, Espresso);
+        System.assertEquals('Espresso', Espresso.type);
+    }
+    @isTest
+    static void brewEspresso_NoWater_ShouldThrowException() {
+        //prepare
+        IEspressoMachine cm = new EspressoMachine();
+        Boolean exceptionThrown = false;
+        //test
+        try{
+            
+            for(Integer i = 0; i < 21; i++) {
```

Comment:
Although it looks like and using a modulus operation always feels cool, I think you should try to avoid making the tests to complicated. less is more :)